pkg/skaffold/constants/constants.go

```diff
@@ -61,8 +61,8 @@ const (
 	DefaultCacheFile   = "cache"
 	DefaultMetricFile  = "metrics"
 
-	DefaultRPCPort     = 50051
-	DefaultRPCHTTPPort = 50052
+	DefaultRPCPort     = 0 // choose an available port at runtime
```

Comment:
would changing the default port from 50051/52 to random be considered as breaking backwards compatibility?
IMO there could be users with scripts or other code that assume these port numbers. Could we not continue trying to assign these ports, but fallback to random only if unavailable?

Reply:
> fallback to random only if unavailable?

that's sort of the problem described in #6425.
silently picking ports will probably life harder for existing integrations.

> would changing the default port from 50051/52 to random be considered as breaking backwards compatibility?

Yes. I opened this per offline chat, but this definitely warrants discussion if we want to do this or not. My assumption was that there are no notable users of these APIs yet but I lack enough context to run with that assumption.